ball.cpp

```diff
@@ -6,18 +6,35 @@ namespace pong
 {
 
 static constexpr float BALL_WIDTH = 25.0f;
-static constexpr float BALL_SPEED = 3.5f;
+static constexpr float BALL_SPEED = 4.5f;
+static int id = 0;
 
 void Ball::OnStart()
 {
     mMesh = std::make_unique<Rectangle>(BALL_WIDTH, BALL_WIDTH);
-    mSpeed = BALL_SPEED;
+    mColliderBox = std::make_unique<ColliderBox>(BALL_WIDTH, BALL_WIDTH);
+    mXSpeed = BALL_SPEED;
+    mYSpeed = BALL_SPEED;
+    mName = "Ball";
 }
 
 void Ball::OnUpdate()
 {
-    mPosition.x += mSpeed;
-    mPosition.y += mSpeed;
+    const glm::vec3 newPosition = GetPosition() + glm::vec3(mXSpeed, mYSpeed, 0.0f);
+    SetPosition(newPosition);
+}
+
+void Ball::OnCollisionStart(GameObject& other)
+{
+    if (other.GetName() == "Player" || other.GetName() == "Opponent"
+     || other.GetName() == "WallV")
```

Comment:
Temporary (maybe?) thing to bouce the ball in the correct direction based on the objects set name. In unity I would normally raycast from the object's previous position to the object it collided with. Then get the normal vector from the raycast and bounce the object that way. But that seems perhaps a little too advanced for right now 😬